File: jschool-spring/src/main/java/com/shak/jschool/entities/course/CourseCommentEntity.java
package com.shak.jschool.entities.course;

import com.shak.jschool.entities.UserEntity;

import javax.persistence.*;

import java.util.Date;

@Entity
public class CourseCommentEntity  {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;


    @ManyToOne(optional = false)
    @JoinColumn(name ="courses_id")
    private CourseEntity course;

    @ManyToOne(optional = false)
    @JoinColumn(name = "users_id")
    private UserEntity user;

    @Column()
    private String comment;

    @Column
    private Date date;


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public CourseEntity getCourseEntity() {
        return course;
    }

    public void setCourseEntity(CourseEntity courseEntity) {
        this.course = courseEntity;
    }

    public UserEntity getUser() {
        return user;
    }

    public void setUser(UserEntity user) {
        this.user = user;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }
}

File: nest-backend/src/courses/comment/course_comment.entity.ts
import {Column, Entity, ManyToOne, PrimaryGeneratedColumn} from "typeorm";
import {User} from "../../users/user.entity";
import {Course} from "../courses.entity";


@Entity()
export class CourseComment {

  @PrimaryGeneratedColumn()
  id: number;

  @ManyToOne(type => Course,{onDelete: 'CASCADE' })
  course:Course;

  @Column()
  courseId:number;

  @Column()
  userId:number;

  @ManyToOne(type => User,{onDelete: 'CASCADE' })
  user:User;

  @Column()
  comment: string;

  @Column()
  date :Date;

}
